zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromIncidentHandler.java

```diff
@@ -66,25 +72,42 @@ public ProcessInstanceForListViewEntity createNewEntity(final String id) {
   public void updateEntity(
       final Record<IncidentRecordValue> record, final ProcessInstanceForListViewEntity entity) {
 
-    final List<List<Long>> elementInstancePath = record.getValue().getElementInstancePath();
-    final List<Integer> callingElementPath = record.getValue().getCallingElementPath();
+    final IncidentRecordValue value = record.getValue();
+    final List<List<Long>> elementInstancePath = value.getElementInstancePath();
+    final List<Integer> callingElementPath = value.getCallingElementPath();
+    final List<Long> processDefinitionPath = value.getProcessDefinitionPath();
 
     final Long processInstanceKey = Long.valueOf(entity.getId());
 
+    // example of how the tree path is build when current instance is on the third level of calling
+    // hierarchy:
+    // PI_<parentProcessInstanceKey>/FN_<parentCallActivityId>/FNI_<parentCallActivityInstanceKey>/
+    // PI_<secondLevelProcessInstanceKey>/FN_<secondLevelCallActivityId>/FNI_<secondLevelCallActivityInstanceKey>PI_<currentProcessInstanceKey>
     final TreePath treePath = new TreePath();
     for (int i = 0; i < elementInstancePath.size(); i++) {
       final List<Long> keysWithinOnePI = elementInstancePath.get(i);
-      if (treePath.isEmpty()) {
-        treePath.startTreePath(keysWithinOnePI.get(0));
-      } else {
-        treePath.appendProcessInstance(keysWithinOnePI.get(0));
-      }
+      treePath.appendProcessInstance(keysWithinOnePI.get(0));
       if (keysWithinOnePI.get(0).equals(processInstanceKey)) {
+        // we stop building the tree path when we reach current processInstanceKey
         break;
       }
-      treePath
-          .appendFlowNode(String.valueOf(callingElementPath.get(i)))
-          .appendFlowNodeInstance(String.valueOf(keysWithinOnePI.get(1)));
+      // get call activity id from processCache
+      final Optional<CachedProcessEntity> cachedProcess =
+          processCache.get(processDefinitionPath.get(i));
+      if (cachedProcess.isPresent()
+          && cachedProcess.get().callElementIds() != null
+          && callingElementPath.get(i) != null) {
+        treePath.appendFlowNode(
+            cachedProcess.get().callElementIds().get(callingElementPath.get(i)));
+      } else {
+        LOGGER.debug(
+            "No process found in cache. TreePath won't contain proper callActivityId. processInstanceKey: {}, processDefinitionKey: {}, incidentKey: {}",
+            processInstanceKey,
+            processDefinitionPath.get(i),
+            record.getKey());
+        treePath.appendFlowNode(String.valueOf(callingElementPath.get(i)));
```

Comment:
❓ This is the ordinal I guess?

Reply:
Yes. We discussed with @mustafadagher , that here I will:
* increase the log level till warning
* will leave the ordinal instead of `callActivityId` here, simply not to loose this information
* will check the code from Operate side, what will be the consequences of having ordinal instead of proper `callActivityId`. I already checked but can't say for sure. Post importer may be affected and also getting statistics for list view diagram badges (`/api/process-instances/statistics` Internal API endpoint). Will add it as todo item in my next task.